backend/typescript/rest/userRoutes.ts

```diff
@@ -119,26 +125,65 @@ userRouter.post("/", createUserDtoValidator, async (req, res) => {
 
 /* Update the user with the specified userId */
 userRouter.put("/:userId", updateUserDtoValidator, async (req, res) => {
+  const userId = Number(req.params.userId);
+  if (Number.isNaN(userId)) {
+    res.status(400).json({ error: "Invalid user ID" });
+    return;
+  }
+
+  const accessToken = getAccessToken(req);
+  if (!accessToken) {
+    res.status(404).json({ error: "Access token not found" });
+    return;
+  }
+
   try {
-    const userId = Number(req.params.userId);
-    if (Number.isNaN(userId)) {
-      res.status(400).json({ error: "Invalid user ID" });
+    const isBehaviourist = await authService.isAuthorizedByRole(
+      accessToken,
+      new Set([Role.ANIMAL_BEHAVIOURIST]),
+    );
+    const behaviouristUpdatableSet = new Set(["skillLevel"]);
```

Comment:
I believe behaviourists can also edit the permissions (can see all logs, can assign users to tasks)